lib/tools/ui/builders/async_child.dart

```diff
@@ -42,3 +43,217 @@ class AsyncChild<T> extends StatelessWidget {
     );
   }
 }
+
+Widget handleLoadingAndError(
+  List<AsyncValue> values,
+  BuildContext context, {
+  Widget Function(BuildContext context)? loadingBuilder,
+  Widget Function(Object? error, StackTrace? stack)? errorBuilder,
+  Widget Function(BuildContext context, Widget child)? orElseBuilder,
+  Color? loaderColor,
+}) {
+  final nonNullOrElseBuilder = orElseBuilder ?? (context, child) => child;
+  if (values.any((value) => value.hasError)) {
+    final nonNullErrorBuilder =
+        errorBuilder ??
+        (error, stack) => Center(
+          child: Text(
+            "${TextConstants.error}:$error",
+            style: TextStyle(color: loaderColor),
+          ),
+        );
+    final firstError = values.firstWhere((value) => value.hasError);
+    final error = firstError.error;
+    final stackTrace = firstError.stackTrace;
+    return nonNullOrElseBuilder(
+      context,
+      nonNullErrorBuilder(error, stackTrace),
+    );
+  }
+  if (values.any((value) => value.isLoading)) {
+    final nonNullLoadingBuilder =
+        loadingBuilder ?? (context) => Loader(color: loaderColor);
+    return nonNullOrElseBuilder(context, nonNullLoadingBuilder(context));
+  }
+  return nonNullOrElseBuilder(context, const SizedBox.shrink());
+}
+
+class Async2Children<P, Q> extends StatelessWidget {
+  final Tuple2<AsyncValue<P>, AsyncValue<Q>> values;
+  final Widget Function(BuildContext context, P value1, Q value2) builder;
+  final Widget Function(Object? error, StackTrace? stack)? errorBuilder;
+  final Widget Function(BuildContext context)? loadingBuilder;
+  final Widget Function(BuildContext context, Widget child)? orElseBuilder;
+  final Color? loaderColor;
+  const Async2Children({
+    super.key,
+    required this.values,
+    required this.builder,
+    this.errorBuilder,
+    this.loaderColor,
+    this.orElseBuilder,
+    this.loadingBuilder,
+  });
+  @override
+  Widget build(BuildContext context) {
+    List<AsyncValue> listValues = [values.item1, values.item2];
+    if (listValues.any((value) => value.hasError || value.isLoading)) {
+      return handleLoadingAndError(
+        listValues,
+        context,
+        loadingBuilder: loadingBuilder,
+        errorBuilder: errorBuilder,
+        orElseBuilder: orElseBuilder,
+        loaderColor: loaderColor,
+      );
+    }
+    return builder(context, listValues[0].value as P, listValues[1].value as Q);
```

Comment:
```suggestion
    return builder(context, values.value1, values.value2);
```

Or maybe there is something I don't get